### PR TITLE
RB-3552: Add question context

### PR DESCRIPTION
**Pull request by elliewem**

### CODE REVIEW

File: source/tryGoal/component/shared/questionContext.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<div class="question-context-inner-wrapper" ng-show="display === 'secondForm' || display === 'firstForm' || display === 'error'">
+    <div class="question-context-response-container" style="margin: 10px;">
+
+        <!--  Technical error -->
+        <div ng-if="display === 'error'">
+            <p>Sorry, but we are unable to show additional information for this question.</p>
```
Comment (captainwillii) [inline source/tryGoal/component/shared/questionContext.html]: Hi Ellie, is this the area where we plan to show detailed error information in the future?  If so, this new panel wouldn't always contain question context information but be a more general 'query context' panel.  What do you think?

Comment (elliewem) [reply source/tryGoal/component/shared/questionContext.html]: Hey Will, I'm not too sure on what we intend this to be in the future but it seems like a good place to provide more detailed error information later on. Shall I change the wording to 'Query Context' to open the scope of it a bit more?

File: source/styles/style.less [hunk 2]
```diff
@@ -190,16 +190,49 @@ span.disabled {
   position: absolute;
   right: 40px;
   top: 10px;
-  color: white;
```
Comment (gregholman) [inline source/styles/style.less]: Out of interest, why the change here?

File: source/tryGoal/component/tryGoalController.js [hunk 3]
```diff
@@ -112,6 +113,28 @@ function($scope, agentMemory, $compile, $stateParams, config, GoalAPI, ConfigAPI
...
+            return concept;
+        }
+    };
+
+    $scope.formatRule = function (rule) {
+        var concepts = rule.match(/\((.*?)\)/gm);
+        var relationship = rule.match(/\[(.*?)\]/);
+
+        concepts[0] = formatConcept(concepts[0]);
+        concepts[1] = formatConcept(concepts[1]);
```
Comment (gregholman) [inline source/tryGoal/component/tryGoalController.js]: Can we be certain that these will always exist?

File: source/styles/style.less (hunk 2 above)
Comment (elliewem) [reply source/styles/style.less]: Mainly for certainty. Just me being picky :)

File: source/tryGoal/component/tryGoalController.js (hunk 3 above)
Comment (elliewem) [reply source/tryGoal/component/tryGoalController.js]: There aren't any checks for response.ruleStack specifically, but we do check whether response.question exists in order to change the display. 
I think it would be super unlikely to not have the rule stack there after checking that the question exists, but it's probably worth having a contingency. I'll wrap it in something :)

File: source/tryGoal/component/tryGoalController.js [hunk 4]
```diff
@@ -112,6 +113,31 @@ function($scope, agentMemory, $compile, $stateParams, config, GoalAPI, ConfigAPI
...
+        }
+    };
+
+    $scope.formatRule = function (rule) {
+        var concepts = rule.match(/\((.*?)\)/gm);
+        var relationship = rule.match(/\[(.*?)\]/);
+
+        if (concepts && relationship) {
+            concepts[0] = formatConcept(concepts[0]);
+            concepts[1] = formatConcept(concepts[1]);
```
Comment (gregholman) [inline source/tryGoal/component/tryGoalController.js]: Sorry, another question 😬 Will concepts and relationships always be of length 2 and will both [0] and [1] always have a value? It looks like you're handling '(null)' - is this the only case to worry about?

Comment (elliewem) [reply source/tryGoal/component/tryGoalController.js]: The rule that's getting parsed is always in this format:
`"Inferring (Ellie)--[should eat at]->(null) (Rule 11/13)"`

There's always the subject and object ((Ellie) and (null)), the relationship ([should eat at]) and the rule number. The ticket describes that the only time we amend the output of one of the concepts is in the case of `null` where we make it a `*` to make it consistent with the training material. Can't think, off the top of my head, of any other times!
